tests/UnitTests.vala

```diff
@@ -3,15 +3,247 @@ namespace Vamp {
         public static int main (string[] args) {
             Test.init (ref args);
 
-            Test.add_func ("/vamp/foo", () => {
-                assert_null (null);
+            Test.add_func ("/vamp/deserialize_basic_config", () => {
+                string package_contents;
+                try {
+                    bool did_open = FileUtils.get_contents (TestConfig.TEST_PACKAGE_FILE, out package_contents);
+                    if (!did_open) {
+                        error ("Failed to get contents from: %s".printf (TestConfig.TEST_PACKAGE_FILE));
+                    }
+
+                    Vamp.Package package = desrialise_package_config (package_contents);
+                    assert_cmpstr (package.name, GLib.CompareOperator.EQ, "test-project");
+                    assert_cmpstr (package.version, GLib.CompareOperator.EQ, "0.0.1");
+                    assert_cmpstr (package.description, GLib.CompareOperator.EQ, "A Test Project");
```

Comment:
I'm in doubt if using the GLib asserts is a good idea, the only one that has real integration with Vala is `GLib.assert()`, the rest will be polluted with the generated code.